runExampleApp.sh

```diff
@@ -65,37 +66,90 @@ function checkForDependencies() {
 }
 
 function updateEnvFile() {
-  paddedMessage "Provide the following based on the Discovery workspace url template:"
-  colorMessage "  https://{cluster_host}:{cluster_port}/discovery/{release_name}/projects/{project_id}/workspace" 3
+  local authType
+  local url
+  local projectId
+  local username
+  local password
+  local apikey
+  local authTypeValid=false
+
+  paddedMessage "What kind of authentication are you using?"
+  echo "iam (Cloud)"
+  echo "cp4d (CP4D)"
   echo
 
-  read -p "  cluster_host: " host
-  read -p "  cluster_port (e.g. 443): " port
-  read -p "  project_id: " projectId
-
-  paddedMessage "CP4D credentials:"
-  echo "  These are the credentials used to log into the CP4D dashboard"
-  echo
-  read -p "  username: " username
-  read -sp "  password: " password
+  while ! $authTypeValid; do
+    read -p "  authType: " authType
+    if [[ "$authType" == "iam" || "$authType" == "cp4d" ]]; then
+      authTypeValid=true
+    else
+      paddedMessage "Please provide a valid value (iam/cp4d)"
+    fi
+  done
 
-  if [[ -z $host || -z $port || -z $username || -z $password ]]; then
+  if [[ "$authType" == "cp4d" ]]; then
+    paddedMessage "CP4D credentials:"
+    echo "  These are the credentials used to log into the CP4D dashboard. Example URL:"
+    colorMessage "  https://zen-25-cpd-zen-25.apps.my-cluster-name.com"
```

Comment:
Does this need ` 3` at the end?

Reply:
https://unix.stackexchange.com/questions/269077/tput-setaf-color-table-how-to-determine-color-codes